optuna/samplers/_gp/_sampler.py

```diff
@@ -0,0 +1,178 @@
+    the acquisition function to suggest the next parameters.
+
+    The current implementation uses:
+        - Matern kernel with nu=2.5 (twice differentiable),
+        - Automatic relevance determination (ARD) for the length scale of each parameter,
+        - Gamma prior for lengthscales, kernel scale, and noise scale,
+        - Log Expected Improvement (logEI) as the acquisition function, and
+        - Quasi-Monte Carlo (QMC) sampling to optimize the acquisition function.
+
+    .. note::
+        This sampler requires `scipy` and `pytorch`.
+        You can install these dependencies with `pip install scipy pytorch`.
+
+    Args:
+        seed:
+            Random seed passed to `independent_sampler`.
+            Defaults to None (random seed is used).
+
+        independent_sampler:
+            Sampler used for initial sampling (for the first `n_startup_trials` trials)
+            and for conditional parameters. Defaults to `None` (a random sampler is used).
+
+        n_startup_trials:
+            Number of initial trials. Defaults to 10.
+    """
+
+    def __init__(
+        self,
+        *,
+        seed: int | None = None,
+        independent_sampler: BaseSampler | None = None,
+        n_startup_trials: int = 10,
+    ) -> None:
+        self._independent_sampler = independent_sampler or optuna.samplers.RandomSampler(seed=seed)
+        self._intersection_search_space = optuna.search_space.IntersectionSearchSpace()
+        self._n_startup_trials = n_startup_trials
+        self._log_prior: "Callable[[_gp.KernelParams], torch.Tensor]" = log_prior
+        self._minimum_noise: float = 1e-6
+        # We cache the kernel parameters for initial values of fitting the next time.
+        self._kernel_params_cache: "_gp.KernelParams | None" = None
+        self._optimize_n_samples: int = 2048
+
+    def reseed_rng(self) -> None:
+        self._independent_sampler.reseed_rng()
+
+    def infer_relative_search_space(
+        self, study: Study, trial: FrozenTrial
+    ) -> dict[str, BaseDistribution]:
+        return self._intersection_search_space.calculate(study)
+
+    def sample_relative(
+        self, study: Study, trial: FrozenTrial, search_space: dict[str, BaseDistribution]
+    ) -> dict[str, Any]:
+        if len(study.directions) > 1:
+            raise NotImplementedError(
+                "GPSampler currently does not support multi-objective optimization."
+            )
+
+        if search_space == {}:
+            return {}
+
+        states = (TrialState.COMPLETE,)
+        trials = study._get_trials(deepcopy=False, states=states, use_cache=True)
+        if len(trials) < self._n_startup_trials:
+            return {}
+
+        (
+            internal_search_space,
+            normalized_params,
+        ) = _search_space.get_search_space_and_normalized_params(trials, search_space)
+        score_vals = np.array([trial.value for trial in trials])
```

Comment:
Btw, I did not pay attention, but how do we handle `inf`, `-inf`, `None`, and `nan`?

Reply:
We don't and cannot handle them, I think. Neither does `BoTorchSampler`.

Reply:
So shall we filter them above?

Reply:
I clamped inf values to current worst/best values.